content/en/blog/_posts/2025-0x-xx-jobs-podreplacementpolicy-goes-ga.md

```diff
@@ -0,0 +1,143 @@
+
+By default, the Job controller immediately recreates Pods as soon as they fail or begin terminating (when they have a deletion timestamp).
+
+As a result, while some Pods are terminating, the total number of running Pods for a Job can temporarily exceed the specified parallelism.
+For Indexed Jobs, this can even mean multiple Pods running for the same index at the same time.
+
+This behavior works fine for many workloads, but it can cause problems in certain cases.
+
+For example, popular machine learning frameworks like TensorFlow and
+[JAX](https://jax.readthedocs.io/en/latest/) expect exactly one Pod per worker index.
+If two Pods run at the same time, you might encounter errors such as:
+```
+/job:worker/task:4: Duplicate task registration with task_name=/job:worker/replica:0/task:4
+```
+
+Additionally, starting replacement Pods before the old ones fully terminate can lead to:
+- Scheduling delays by kube-scheduler as the nodes remain occupied.
+- Unnecessary cluster scale-ups to accommodate the replacement Pods.
+- Temporary bypassing of quota checks by workload orchestrators like [Kueue](https://kueue.sigs.k8s.io/).
+
+With Pod replacement policy, Kubernetes gives you control over when the control plane
+replaces terminating Pods, helping you avoid these issues.
+
+## How Pod Replacement Policy works
+
+This enhancement means that Jobs in Kubernetes have an optional field `.spec.podReplacementPolicy`.  
+You can choose one of two policies:
+- `TerminatingOrFailed` (default): Replaces Pods as soon as they start terminating.
+- `Failed`: Replaces Pods only after they fully terminate and transition to the `Failed` phase.
+
+Setting the policy to `Failed` ensures that a new Pod is only created after the previous one has completely terminated.
+
+For Jobs with a Pod Failure Policy, the default `podReplacementPolicy` is `Failed`, and no other value is allowed.
+See [Pod Failure Policy](/docs/concepts/workloads/controllers/job/#pod-failure-policy) to learn more about Pod Failure Policies for Jobs.
+
+You can check how many Pods are currently terminating by inspecting the Job’s `.status.terminating` field:
+
+```sh
+kubectl get job myjob -o=jsonpath='{.status.terminating}'
+```
+
+## Example
+
+Here’s a Job example that executes a task two times  (`spec.completions: 2`) and replaces Pods only after they fully terminate (`spec.podReplacementPolicy: Failed`):
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: example-job
+spec:
+  completions: 2
+  podReplacementPolicy: Failed
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: worker
+        image: your-image
+```
+
+If a Pod receives a SIGTERM signal, it begins terminating.
+When the container handles termination gracefully, cleanup may take some time.
+
+With the `TerminatingOrFailed` policy, we would have the following sequence of events:
+```sh
+kubectl get pods --watch
+
+NAME                READY   STATUS              RESTARTS   AGE
```

Comment:
In general the `--watch` output is hard to follow, unless you're doing that in a live cluster. So I'd suggest to replace that with a series of simple gets, which will present 2 runnings pods first, in the 2nd output 3 pods, 2 running and 1 terminated. Something like the following:
```
NAME                READY   STATUS              RESTARTS   AGE
Comment
example-job-t9x2r   1/1     Running             0          1s
example-job-f9xrq   1/1     Running             0          0s
```

```
NAME                READY   STATUS              RESTARTS   AGE
Comment
example-job-t9x2r   1/1     Failed         0          6s
example-job-f9xrq   1/1     Running             0          0s
example-job-t9x2r   1/1     Running           0          21s
```